src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+
+    dist = evaluate(c.distance, p, uniform)
+    dist *= h
+
+    return dist
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractArray)
+    p = probabilities(x, c.est)
+    L = alphabet_length(c)
+    h = entropy(c.entropy, p) / log(c.entropy.base, L)
+    complexity(c, p; h)
+end
+
+function complexity_normalized(c::StatisticalComplexity, p::Probabilities; h::Float64)
+    comp = complexity(c, p; h)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+function complexity_normalized(c::StatisticalComplexity, x::AbstractArray)
+    comp = complexity(c, x)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+linearpermissiverange(start; stop, length) = length==1 ? (start:start) : range(start, stop=stop, length=length)
+
+"""
+    maximum_complexity_entropy(c::StatisticalComplexity; num=1)
+
+Calculate the maximum complexity-entropy curve for the statistical complexity according to [^Rosso2007]
+for `num * m!` different values of the normalized permutation entropy.
+
+## Description
+The way the statistical complexity is designed, there is a minimum and maximum possible complexity
+for data with a given permutation entropy.
+The calculation time of the maximum complexity curve grows as O((m!)^2), and thus takes very long for higher `m`.
+
+This function is adapted from S. Sippels implementation in statcomp [^statcomp].
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102.
+
+[^statcomp] Sippel, S., Lange, H., Gans, F. (2019).
+            [statcomp: Statistical Complexity and Information Measures for Time Series Analysis](https://cran.r-project.org/web/packages/statcomp/index.html)
+"""
+function maximum_complexity_entropy(c::StatisticalComplexity; num::Int=1)
+    
+    L = alphabet_length(c)
+    # in these we'll write the entropy (h) and corresponding max. complexity (c) values
+    hs, cs = zeros(L-1, num), zeros(L-1, num)
+    norm = log(c.entropy.base, L)
+    for i in 1:L-1
+        p = zeros(L)
+        prob_params = linearpermissiverange(0; stop=1/L, length=num)
+        for k in 1:num
+            p[1] = prob_params[k]
+            for j in 1:L-i
+                p[j] = (1-prob_params[k]) / (L-i)
+            end
```

Comment:
What's going on here? You first assign `p[1]`, but then you iterate over `j` starting from 1 so you assign `p[1]` again?

Reply:
Ah, this is a result of the fact that I didn't pay attention while translating this! Fascinatingly, the tests still passed